### ISSUE TITLE
[Feature Request] Setting to prevent history rewriting for remote commits

### ISSUE THREAD

**Issue by markmsmith** [git-up/GitUp#217]
For users that want to avoid accidentally rewriting history on commits that have been pushed to a remote repo (and messing up all the project collaborators), it would be helpful if there was a user preference for "disable remote history rewrites".  This would check, on a per-commit basis, if a given action would result in a modification to any commits that had already been pushed to a remote repo.  If it would, the action is disabled.
I know there's already the functionality around force pushes etc, but it seems like it would be better to prevent the change in the first place, rather than it only being discovered at push time, potentially after several changes.

I don't know how easy the codebase makes it to track whether a given commit has been pushed, or the ripple effects for a given edit, but it would be really nice to distinguish history that is "public record" vs what you can play with locally, and have different rules for the local stuff.

**Comment by swisspol** [comment]
Note that by definition the Map view will show you if a commit you're rewriting is on a branch whose tip is on the server e.g. its tip label says `origin/master`.

**Comment by brassel** [comment]
I would like to back Marks request.
Why would this be useful? 
While it is great that gitup enables history editing so easily it nearly makes it unusable in my case. I found gitup while searching for a simple git client to collaborate with less nerdy people like our graphic designers and film effects crew. So I tell them about the basic usage of gitup and they get it working real quick. But after a time they always call me:
"I have tried this and that and see how this commit line thingy looks like now! And I can't pull or push anymore."
And I have to tell them that they should not use this feature and that feature and it is really hard to explain why these features are there in the first place when they should not use them.
So, to cut to the chase: Disabling this stuff for already pushed parts would really help.

**Comment by dohnuts** [comment]
+1

rewriting history is inherently bad , the history is not made up...unless it is rewritten

**Comment by sesam** [comment]
Yes it makes sense to not allow modifying remote branches locally. Maybe gitup could automatically create a local branch and use that instead when the user tries to do this?
